crates/transaction-emitter-lib/src/args.rs

```diff
@@ -82,6 +125,9 @@ pub struct EmitArgs {
 
     #[clap(long, help = "Percentage of invalid txs", default_value = "0")]
     pub invalid_tx: usize,
+
+    #[clap(long, default_value_t = TransactionType::P2P)]
```

Comment:
Add `value_enum` here and you won't need to impl FromStr yourself, and it'll restrict the values to the possible enum values.

Example code: https://gist.github.com/banool/82d0d39a4b95b56973891f662abe3a45.